decorator.rb

```diff
@@ -0,0 +1,24 @@
+require_relative 'nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    super.strip
+  end
```

Comment:
- [ ] Great job creating TrimmerDecorator class, 👏🏻 However it does not work as expected. According to the project requirements, The TrimmerDecorator class should check if the name has a maximum of 10 characters. If it's longer it should trim the word.
- TrimmerDecorator class should look similar to:

```ruby
class TrimmerDecorator < Decorator
    def correct_name
      @nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name
    end
end
```

- Here is your Final result:

![image](https://user-images.githubusercontent.com/79058364/231302432-33d13dba-8cec-44f2-9633-72e0405aebd7.png)